x-pack/plugins/observability_solution/apm/public/components/shared/try_it_button/index.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBetaBadge,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiLoadingSpinner,
+  EuiPopover,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+
+interface Props {
+  isFeatureEnabled: boolean;
+  promoLabel?: string;
+  linkLabel: string;
+  onClick: () => void;
+  popoverContent?: React.ReactElement;
+  icon?: 'beta' | 'beaker';
+  isLoading: boolean;
+}
+
+export function TryItButton({
+  isFeatureEnabled,
+  linkLabel,
+  onClick,
+  popoverContent,
+  promoLabel,
+  icon,
+  isLoading,
+}: Props) {
+  const { core } = useApmPluginContext();
+  const canEditAdvancedSettings =
+    core.application.capabilities.advancedSettings?.save;
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  function togglePopover() {
+    setIsPopoverOpen((state) => !state);
+  }
+
+  function TryItBadge() {
```

Comment:
Maybe this function can be moved from this component as it's not relying on the state and maybe can be reused as a component, wdyt?

Reply:
I don't think it's necessary at this point, especially since this is wrapped inside a `EuiFlexItem` and it doesn't make much sense to export a function with this. And if I remove the `EuiFlexItem` component it's just a `EuiBetaBadge`. I'm kind of against micro-components.